src/components/EditContact.js

```diff
@@ -0,0 +1,162 @@
+import { useState } from "react"
+import { useNavigate, useParams } from "react-router-dom";
+import { useEffect } from 'react'
+
+function EditContact(props) {
+
+  let navigate=useNavigate()
+
+  // setContacts and contacts must be passed as props
+  // to this component so new contacts can be added to the
+  // state
+  const { setContacts, contacts } = props
+
+  //TODO: Implement controlled form
+  //send POST to json server on form submit
+
+  //Right here, I am initializing the firstName, lastName, street and city to a blank columm
+
+const [firstName, setFirstName]=useState('')
+const [lastName, setLastName]=useState('')
+const [street, setStreet]=useState('')
+const [city, setCity]=useState('')
+const [email, setEmail]=useState('')
+const [linkedin, setLinkedin]=useState('')
+const [twitter, setTwitter]=useState('')
+
+const params=useParams()
+
+console.log('Value of Params', params)
+useEffect(()=>{
+fetch(`http://localhost:4000/contacts/${params.id}`)
+.then(res=> res.json())
+.then(json=>{
+  console.log(json)
+  setFirstName(json.firstName)
+  setLastName(json.lastName)
+  setStreet(json.street)
+  setCity(json.city)
+  setEmail(json.email)
+  setLinkedin(json.linkedin)
+  setTwitter(json.twitter)
+})
+}, [params])
+
+//I am writing what should be implemented after a submit button has been clicked in the onSubmit function.
+function onSubmit(event) {
+event.preventDefault()
+// console.log(firstName, lastName, street, city, email, linkedin, twitter)
+
+const options = {
+  method:'PUT',
+  headers: {
+    'Content-Type': 'application/json'
+  },
+  body:JSON.stringify({
+    firstName:firstName,
+    lastName:lastName,
+    street:street,
+    city:city,
+    email:email,
+    linkedin:linkedin,
+    twitter:twitter
+  }),
+}
+
+
+
+fetch(`http://localhost:4000/contacts/${params.id}`, options)
+.then((res)=>res.json())
+.then ((json) =>{
+  navigate('/')
+  console.log('contacts created', json)
```

Comment:
I like that you are constantly console logging everything you get a response from the json server to check that what you want renders is displaying in the console. Good work!